ownCloud/Settings/LogSettingsViewController.swift

```diff
@@ -139,63 +139,13 @@ class LogSettingsViewController: StaticTableViewController {
-							alert.addAction(UIAlertAction(title: "Cancel".localized, style: .default, handler: nil))
-
-							if !logFileWriter.enabled {
-								alert.addAction(UIAlertAction(title: "Enable log file".localized, style: .default, handler: { (_) in
-									logFileWriter.enabled = true
-									logFileWriterSwitchRow?.value = logFileWriter.enabled
-								}))
-							}
-
-							self?.present(alert, animated: true, completion: nil)
-						} else {
-							let logURL = FileManager.default.temporaryDirectory.appendingPathComponent("ownCloud App Log.txt")
-
-							do {
-								if FileManager.default.fileExists(atPath: logURL.path) {
-									try FileManager.default.removeItem(at: logURL)
-								}
-
-								try FileManager.default.copyItem(at: logFileWriter.logFileURL, to: logURL)
-							} catch {
-							}
-
-							let shareViewController = UIActivityViewController(activityItems: [logURL], applicationActivities:nil)
-							shareViewController.completionWithItemsHandler = { (_, _, _, _) in
-								do {
-									try FileManager.default.removeItem(at: logURL)
-								} catch {
-								}
-							}
-
-							if UIDevice.current.isIpad() {
-								shareViewController.popoverPresentationController?.sourceView = row.cell
-								shareViewController.popoverPresentationController?.sourceRect = CGRect(x: row.cell?.bounds.midX ?? 0, y: row.cell?.bounds.midY ?? 0, width: 1, height: 1)
-								shareViewController.popoverPresentationController?.permittedArrowDirections = .down
-							}
-
-							row.viewController?.present(shareViewController, animated: true, completion: nil)
-						}
-					}
-				}, title: "Share log file".localized, style: .plain, identifier: "share-logfile"))
-
-				logOutputSection?.add(row: StaticTableViewRow(buttonWithAction: { (row, _) in
-					let alert = UIAlertController(with: "Really reset log file?".localized, message: "This action can't be undone.".localized, destructiveLabel: "Reset log file".localized, preferredStyle: .alert, destructiveAction: {
-						OCLogger.shared.pauseWriters(intermittentBlock: {
-							if let logFileWriter = OCLogger.shared.writer(withIdentifier: .writerFile) as? OCLogFileWriter {
-								logFileWriter.eraseOrTruncate()
-							}
-						})
-					})
-
-					row.viewController?.present(alert, animated: true, completion: nil)
-				}, title: "Reset log file".localized, style: .destructive, identifier: "reset-logfile"))
+				// Creation of the frequency row.
+				let logsRow = StaticTableViewRow(subtitleRowWithAction: { [weak self] (_, _) in
+						let logFilesViewController = LogFilesViewController()
+						self?.navigationController?.pushViewController(logFilesViewController, animated: true)
+					}, title: "Log Files".localized, accessoryType: .disclosureIndicator, identifier: "viewLogs")
+				logOutputSection?.add(row: logsRow)
+				logOutputSection?.footerTitle = "Up to 10 recent logs are kept on the device and each log can accumulate messages for the last 24 hours of the app use".localized
```

Comment:
Since the File Provider also generates log output if logging is enabled, I'd like to suggest this instead:
"The last 10 log files are kept on the device - with each log file covering up to 24 hours of usage. You can export and delete log files by selecting "Browse log files"."

Regarding the "Mask private data" option, I noticed there's already some text pointing out its limitations, so I guess no change is required there.

Adding a notice below the "Enable logging" toggle might be a good idea, though, pointing out that log files
- may end up containing sensitive information
- can impact performance
- are stored on the device and not sent to ownCloud unless explicitly shared by the user themselves.
- … anything else? @michaelstingl @hosy

Reply:
@felix-schwarz I changed the first part.. And regarding adding another note to "Enable Logging" switch, I came up with following text:

_When activated, logs may impact performance and include sensitive information. However the logs are not subject to automatic submission to ownCloud servers. Sharing logs with others is sole user responsibility._